trieste/acquisition/function/multi_objective.py

```diff
@@ -255,13 +254,21 @@ def prepare_acquisition_function(
         # hypervolume improvement in this area
         _partition_bounds = prepare_default_non_dominated_partition_bounds(_reference_pt, _pf.front)
 
-        sampler = BatchReparametrizationSampler(self._sample_size, model)
+        try:
+            sampler = model.reparam_sampler(self._sample_size)
```

Comment:
if we implement the `ReparametrizationSamplerModel` interface, then we can also be more precise above in specifying the type of the probabilistic model